torchgeo/datasets/ssl4eo_downstream_landsat.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Self-Supervised Learning for Earth Observation Downstream Evaluation."""
+
+import glob
+import os
+from typing import Callable, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from torch import Tensor
+
+from .geo import NonGeoDataset
+
+
+class SSL4EODownstream(NonGeoDataset):
+    """SSL4EO Downstream Evaluation Dataset.
+
+    Dataset is intended to be used for evaluation of SSL techniques.
+
+    Dataset format:
+
+    * input landsat image and single channel mask
+
+    Each patch has the following properties:
+
+    * 264 x 264 pixels
+    * Resampled to 30 m resolution (7920 x 7920 m)
+    * Single multispectral GeoTIFF file
+
+    .. versionadded:: 0.5
+    """
+
+    valid_input_sensors = ["tm_toa", "etm_toa", "etm_sr", "oli_tirs_toa", "oli_sr"]
+    valid_mask_products = ["cdl", "nlcd"]
+    valid_splits = ["train", "val", "test"]
+
+    image_root = "ssl4eo_l_*_benchmark"
```

Comment:
Benchmark or downstream? We should be consistent with file/directory/class names to avoid confusion.